src/modules/Modal/Modal.js

```diff
@@ -133,8 +148,13 @@ class Modal extends Component {
       }
     }
 
-    document.addEventListener('keydown', this.handleDocumentKeyDown)
-    document.addEventListener('click', this.handleClickOutside)
+    if (closeOnEscape) {
+      document.addEventListener('keydown', this.handleDocumentKeyDown)
+    }
```

Comment:
To clarify, let's say we add listeners on open only if the `onClose*` props are true.

If a Modal opens and later, while open, then `closeOnEscape` becomes true. Then we need to add a condition to componentDidUpdate `if !prevProps.closeOnEscape && this.propscloseOnEscape then add listener`.  But, we don't really know if the listener was already added either :S We can assume so, if we put all the logic in place elsewhere to make it so.  But it is convoluted.

This is part of a larger issue, we have a pending RFC to implement a better way to handle events.  Handle pooling and delegation.  See #284.